### DISCUSSION TITLE
⚡ Xcode 16

### DISCUSSION THREAD

**Discussion by helinanever** [codemagic-ci-cd/codemagic-docs#2594] [Announcements]
**Xcode 16.0 Beta 1** is now available in Codemagic along with new runtimes for iOS 18.0, watchOS 11.0, tvOS 18.0, and visionOS 2.0. To use it, set the Xcode version to `16` or `edge` in your build configuration.

See more details about the build machine specs and preinstalled software in our [documentation](https://docs.codemagic.io/specs/versions-macos/).

**Comment by helinanever** [comment]
The build machine image has been updated with **Xcode 16.0 Beta 2**.

**Comment by helinanever** [comment]
The build machine image now has **Xcode 16.0 Beta 3** available.

**Comment by mikhail-tokarev** [comment]
The build machine image now has **Xcode 16.0 Beta 4** available.

**Comment by helinanever** [comment]
The build machine image now has **Xcode 16.0 Beta 5** available.

**Comment by helinanever** [comment]
The build machine image now has **Xcode 16.0 Beta 6** available.

**Comment by mikhail-tokarev** [comment]
The build machine image now has **Xcode 16.0 RC** available.

**Reply by adamrushy** [reply]
Hey, 👋🏼 

It looks like the current installed image only has the iPhone SE available as a simulator.

> ERROR [2024-09-13 21:38:57.64]: Ignoring 'iPhone 16', couldn't find matching simulator
> ERROR [2024-09-13 21:38:57.64]: Couldn't find any matching simulators for '["iPhone 16"]' - falling back to default simulator
> INFO [2024-09-13 21:38:57.64]: Found simulator "iPhone SE (3rd generation) (18.0)"

**Reply by helinanever** [reply]
Hey @adamrushy, the image has the following iPhone simulators available:
iPhone 15
iPhone 15 Plus
iPhone 15 Pro
iPhone 15 Pro Max
iPhone 16
iPhone 16 Plus
iPhone 16 Pro
iPhone 16 Pro Max
iPhone SE (3rd generation) 

If you run into the same issue again, please share your build ID so we can look into it.

**Reply by adamrushy** [reply]
Hey, @helinanever 

I gave it another go, but I am still getting the same issue:

> DEBUG [2024-09-16 08:38:30.30]: Fetching available simulator devices
nil versions are discouraged and will be deprecated in Rubygems 4
> ERROR [2024-09-16 08:38:32.70]: Ignoring 'iPhone 16', couldn't find matching simulator
> ERROR [2024-09-16 08:38:32.70]: Couldn't find any matching simulators for '["iPhone 16"]' - falling back to default simulator
> INFO [2024-09-16 08:38:32.70]: Found simulator "iPhone SE (3rd generation) (18.0)"

Build ID: 66e7ee0fed9a7fe8f05431e3

**Reply by himesh-cm** [reply]
Hey @adamrushy,
I have initiated the conversation on intercom, let's take our discussion on intercom.

**Reply by himesh-cm** [reply]
As discussed, setting device type like this along with OS version as shown below in your fastlane file works as expected.
`device: iPhone 16 (18.0)`

**Comment by jonmountjoy** [comment]
I get asset validation errors when I switched to Xcode 16 (a few hours ago).  When I perform the same compilation on my Mac (recently upgraded to the latest OS and Xcode 16.0 (16A242d) it works just fine.  (When using the Organizer to send it manually).

Using CodeMagic, I get the error noted below.  Has anyone else encountered this?

<details><summary> Asset validation failed (90534) Unsupported SDK or Xcode version </summary>
<p>
2024-09-17 13:28:04.286 * Error: [ContentDelivery.Uploader.600000EF4180] Asset validation failed (90534) Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn’t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https://developer.apple.com/news/releases. (ID: 20414e33-eb0f-4a6b-9148-57ab44f985d5)
2024-09-17 13:28:07.357 [ContentDelivery.Uploader.600000EF4180]   

=============. 
UPLOAD FAILED with 1 error.  
=============.  

{"tool-version":"8.003.16003","tool-path":"\/Applications\/Xcode-16.0.app\/Contents\/SharedFrameworks\/ContentDeliveryServices.framework\/Versions\/A\/Frameworks\/AppStoreService.framework","os-version":"14.6.1","product-errors":[{"message":"Asset validation failed","userInfo":{"NSUnderlyingError":"Error Domain=IrisAPI Code=-19241 \"Asset validation failed\" UserInfo={status=409, detail=Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn’t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https:\/\/developer.apple.com\/news\/releases., id=20414e33-eb0f-4a6b-9148-57ab44f985d5, code=STATE_ERROR.VALIDATION_ERROR.90534, title=Asset validation failed, NSLocalizedFailureReason=Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn’t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https:\/\/developer.apple.com\/news\/releases., NSLocalizedDescription=Asset validation failed}","NSLocalizedDescription":"Asset validation failed","iris-code":"STATE_ERROR.VALIDATION_ERROR.90534","NSLocalizedFailureReason":"Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn’t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https:\/\/developer.apple.com\/news\/releases. (ID: 20414e33-eb0f-4a6b-9148-57ab44f985d5)"},"code":90534}]}


</p>
</details>

**Reply by helinanever** [reply]
Hi @jonmountjoy, it might be that we had not completed the release version rollout and still had the RC version on the build machines at the time you ran your build. However, we have now updated all build machines with build `16A242d`.

**Reply by jonmountjoy** [reply]
Thanks @helinanever - I can confirm that it's all now working just fine.  
I notice no change at all in the UI - and it may be useful to somehow indicate when a version is a release version or not (or something).

**Reply by helinanever** [reply]
Thanks for this feedback, we've taken note of it.

**Reply by helinanever** [reply]
You can always check the build machine specs and the exact Xcode version in our [documentation](https://docs.codemagic.io/specs/versions-macos/) which gets updated as soon as the rollout is complete.

**Reply by jonmountjoy** [reply]
Ah that's great information - thank you.

**Comment by helinanever** [comment]
We now have the **Xcode 16.0 release** version available on the build image.

**Reply by jkoenig134** [reply]
If I want to use this I have to switch from `xcode: latest` to `xcode: 16.0` in my `codemagic.yaml` file right? https://docs.codemagic.io/specs/versions-macos/ is not really helpful on how to actually choose the XCode Version using the yaml file.

**Reply by helinanever** [reply]
Correct. The alias `latest` currently still points to `15.4` and will be switched to `16.0` later. We will announce the switch date ahead of time. However, thank you for this feedback, we will think of ways to improve the docs. :)

**Reply by jkoenig134** [reply]
thanks!

**Comment by Nodirbek-Khudoyberdiev** [comment]
xcode 16 error while uploading a build to app store connect

"Step 3 script Set up provisioning profiles settings on Xcode project exited with status code 1"
build id -> 6713da15c395237e8dabea28
please help
urgent
@helinanever
